### PR TITLE
[functions] Use std lib zipfile to check if python function is a .py or a .zip

### PR DESCRIPTION
**Pull request by mauza**
When running a python function in pulsar on a broker don't rely on the extension to tell if the file is a zipfile or not but use the built in zipfile.is_zipfile(filepath) function to check.

### Motivation

Automation using the pulsar admin api saves python zip files as .py files which breaks when running the function on a broker because python_instance_main.py only checks if they file has a .zip extension for it to extract all the contents.

### Modifications

Changed the conditional that checks for a .zip extensions to now use zipfile.is_zipfile

### Verifying this change

- [?] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.  <---- I'm guessing

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (don't know)

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by mauza** [comment]
After testing with this change locally it doesn't seem to be helping my problem. Closing PR.

**Comment by mauza** [comment]
I actually think this still may be a fix.

File: pulsar-functions/instance/src/main/python/python_instance_main.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def main():
       zpfile = zipfile.ZipFile(str(args.py), 'r')
       zpfile.extractall(os.path.dirname(str(args.py)))
     sys.path.insert(0, os.path.dirname(str(args.py)))
-  elif os.path.splitext(str(args.py))[1] == '.zip':
+  elif zipfile.is_zipfile(str(args.py)):
```
Comment (zymap) [inline pulsar-functions/instance/src/main/python/python_instance_main.py:116]: can we add a test for this?

Comment (mauza) [reply pulsar-functions/instance/src/main/python/python_instance_main.py:116]: Yeah, I'm new to contributing so any direction on how to go about doing that would be helpful, but I'll dive in see if I can figure out how to create some tests around this.

Comment (zymap) [reply pulsar-functions/instance/src/main/python/python_instance_main.py:116]: You can take a look at `test_python_instance.py`.

**Comment by codelipenghui** [comment]
The pr had no activity for 30 days, mark with Stale label.

File: pulsar-functions/instance/src/main/python/python_instance_main.py (hunk 1 above)
Comment (tisonkun) [reply pulsar-functions/instance/src/main/python/python_instance_main.py:116]: @zymap I think we don't test 3rd-party dependency for their own functionality :)

**Comment by tisonkun** [comment]
It seems the author deleted the patch branch.

Closed as stale.

**Comment by tisonkun** [comment]
Although, I think it's good to have. So if anyone continue this patch, I'm glad to review.

**Pull request closed without merging**